### PR TITLE
Add version number

### PR DESCRIPTION
**Pull request by yourslab**
Add version number and missing `@file` doxygen tag to `shadow_cbmc_state.c`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/include/shadow.h [hunk 1]
```diff
@@ -1,4 +1,5 @@
 /*
+ * Device Shadow Library V1.0.0
```
Comment (yourslab) [inline source/include/shadow.h:2]: Is this the right name? Should it be `AWS IoT Device Shadow Library V1.0.0`?

Comment (sarenameas) [reply source/include/shadow.h:2]: This is a PM question

Comment (yngki) [reply source/include/shadow.h]: according to Tanmoy:  AWS IoT Device Shadow V1.0.0

**Review by yngki — APPROVED** [review]
Library name confirmed with Tanmoy

**Review by abhidixi11 — APPROVED** [review]

**Review by sbammi — APPROVED** [review]

**Review by cobusve — APPROVED** [review]

**Pull request merged**